### DISCUSSION TITLE
Deleted OSM data still displays on the vector map

### DISCUSSION THREAD

**Discussion by xlam** [osmandapp/OsmAnd#11763] [Q&A] [answered]
Hello, everybody!
I have edited a small piece of OSM data, and I'm confused about how it is rendered on the vector map. I just deleted the wrong placed highway and drawn a new one just where it should be. Surprisingly, the deleted piece of highway is still shown after the map update along with the new one. OSM and other tile maps display it correctly (see here: [https://www.openstreetmap.org/#map=19/51.55529/45.92930](https://www.openstreetmap.org/#map=19/51.55529/45.92930)) Am I missing something, or doing something wrong? Any ideas?
![osmand-bug2](https://user-images.githubusercontent.com/179263/119533391-d74fb000-bd96-11eb-9558-e36044a88627.jpg)
![osm1](https://user-images.githubusercontent.com/179263/119533075-8c359d00-bd96-11eb-8e4b-82f4d971f6c5.jpg)

**Comment by ivanPyrohivskyi** [comment, marked as answer]
These can be rendering cache. Just close and open the app again.
We checked it and live updates works correct.

**Reply by xlam** [reply]
Thanks for the answer, and sorry for my incompetence ;) There weren't any issues with rendering cache. Actually, there was local roads file for my region that had been rendered along with the live-updated map. It was not so obvious that the local roads file and the main map file will be displayed at the same time. And I thought that the roads file also receive live updates. Anyway, I just deleted local roads and old geometry has disappeared. Thank OsmAnd team for the great app!

**Comment by Lee-Carre** [comment]
Aside from OsmAnd, re editing OSM elements, instead of deleting & re-adding, it's better to modify. See [OSM Wiki § Keep the history](//wiki.openstreetmap.org/wiki/Keep_the_history).
